plugins/newsdownloader.koplugin/main.lua

```diff
@@ -66,15 +66,55 @@ function NewsDownloader:init()
     self.ui.menu:registerToMainMenu(self)
 end
 
+local function getWifiEnableAction()
+   local wifi_enable_action_setting = G_reader_settings:readSetting("wifi_enable_action") or "prompt"
+   local wifi_enable_actions = {
+       turn_on = {_("turn on"), _("Turn on")},
+       prompt = {_("prompt"), _("Prompt")},
+       no_action = {_("no action"), _("Do nothing")},
+   }
+   local action_table = function(wifi_enable_action)
+       return {
+           text = wifi_enable_actions[wifi_enable_action][2],
+           checked_func = function()
+               return wifi_enable_action_setting == wifi_enable_action
+           end,
+           callback = function()
+               wifi_enable_action_setting = wifi_enable_action
+               G_reader_settings:saveSetting("wifi_enable_action", wifi_enable_action)
+           end,
+       }
+   end
+   return {
+       text_func = function()
+           return FFIUtil.template(
+               _("Action when Wi-Fi is off: %1"),
+               wifi_enable_actions[wifi_enable_action_setting][1]
+           )
+       end,
+       sub_item_table = {
+           action_table("turn_on"),
+           action_table("prompt"),
+           action_table("no_action"),
+       }
+   }
+end
+
 function NewsDownloader:addToMainMenu(menu_items)
     self:lazyInitialization()
-
     menu_items.news_downloader = {
         text = _("News (RSS/Atom) downloader"),
         sub_item_table = {
             {
                 text = _("Download news"),
-                callback = function() self:loadConfigAndProcessFeeds() end,
+                callback = function()
+                    if not NetworkMgr:isOnline() then
+                        wifi_enabled_before_action = false
+                        NetworkMgr:wifiEnableAction(loadConfigAndProcessFeeds)
```

Comment:
In this case, as this is the last command of a callback, one could just use here `Trapper:wrap( function() NetworkMgr:wifiEnableAction(loadConfigAndProcessFeeds) end )` to be able to use the other Trapper methods in wifiEnableAction, which will call NetworkMgr:turnOnWifiAndWaitForConnection()... but it's all quote convoluted...

Reply:
We can probably do without to start with, but 60 seconds is significantly too long in that case. 10 seconds? 5?

Reply:
@Frenzie you're talking about [`local timeout = 60;`](https://github.com/koreader/koreader/pull/3482/files#diff-8a665adb55900a011d1c3928d7447e61R66)
If you had to wait for wifi to be up, then select network, type password, wait till connection is established - then 5 sec is definitely too small 10 also. I'll decrease it to 30 sec.

Reply:
Trapper seems to be a big change, too complicated. 
How about marking this as TODO for next PR?

Reply:
Perhaps I'm misunderstanding, but if I accidentally trigger a wifi-related action pretty much anywhere but at home I don't want to be stuck for 60 seconds waiting for some one-minute timer to run out. ;-)